src/mame/drivers/dfs500.cpp

```diff
@@ -0,0 +1,1036 @@
+	// TODO: m_adc->in_callback<3>().set_ioport("XCOM0");
+	// TODO: m_adc->in_callback<2>().set_ioport("YCOM0");
+	// TODO: m_adc->in_callback<1>().set_ioport("TCLIP2");
+	// TODO: m_adc->in_callback<0>().set_ioport("TCLIP1");
+
+	//Buzzer		
+	SPEAKER(config, "mono").front_center();
+	BEEP(config, "buzzer", 4000).add_route(ALL_OUTPUTS, "mono", 0.05); // incorrect/arbitrary freq.
+	                                                                   // I did not calculate the correct one yet.
+
+	/******************* Effects Processing Unit ********************/
+	// CXQ70116P-10 at IC40 (same as V20, but with a 16-bit data bus)
+	V30(config, m_maincpu, 8_MHz_XTAL);
+	m_maincpu->set_addrmap(AS_PROGRAM, &dfs500_state::maincpu_mem_map);
+	m_maincpu->set_irq_acknowledge_callback("pic", FUNC(pic8259_device::inta_cb));
+
+	// CXQ70116P-10 at IC9
+	V30(config, m_effectcpu, 8_MHz_XTAL);
+	m_effectcpu->set_addrmap(AS_PROGRAM, &dfs500_state::effectcpu_mem_map);
+
+	// CXQ71054P at IC51 (Programmable Timer / Counter)
+	PIT8254(config, m_pit, 0);
+	m_pit->set_clk<0>(8_MHz_XTAL);
+	m_pit->set_clk<1>(8_MHz_XTAL);
+	m_pit->out_handler<1>().set(m_pit, FUNC(pit8254_device::write_clk2));
+	m_pit->out_handler<0>().set(m_serial1, FUNC(i8251_device::write_txc));
+	m_pit->out_handler<0>().append(m_serial1, FUNC(i8251_device::write_rxc));
+	m_pit->out_handler<0>().append(m_serial2, FUNC(i8251_device::write_txc));
+	m_pit->out_handler<0>().append(m_serial2, FUNC(i8251_device::write_rxc));
+
+	// NEC D71059C at IC52 (Programmable Interruption Controller)
+	PIC8259(config, m_pic, 0);
+	m_pic->out_int_callback().set_inputline(m_maincpu, 0);
+
+	// CXQ71051P at IC53 (Serial Interface Unit)
+	I8251(config, m_serial1, 8_MHz_XTAL);
+	m_serial1->txd_handler().set(m_cpanel_serial, FUNC(i8251_device::write_rxd));
+	m_serial1->txrdy_handler().set(m_pic, FUNC(pic8259_device::ir7_w));
+
+	// CXQ71051P at IC54 (Serial Interface Unit)
+	I8251(config, m_serial2, 8_MHz_XTAL);
+	m_serial2->txrdy_handler().set(m_pic, FUNC(pic8259_device::ir6_w));
+	// FIXME: Declare an interface to hook this up to another emulated device in MAME (such as pve500)
+	//
+	// This goes to the CN21 EDITOR D-SUB CONNECTOR on board CN-573
+	// I think the purpose of this is to connect to an editor such as the Sony PVE-500.
+	//
+	// m_serial2->txd_handler().set(m_..., FUNC(..._device::write_txd)); "XMIT"
+	// m_...->rxd_handler().set(m_serial2, FUNC(i8251_device::write_rxd)); "RCV"
+
+	GENERIC_LATCH_16(config, "controlsignals");
+	GENERIC_LATCH_16(config, "reg7_5");
+	GENERIC_LATCH_16(config, "xflt");
+	GENERIC_LATCH_16(config, "yflt");
+
+	// In the future this could become IMAGE_AVIVIDEO (or even, perhaps, we
+	// should add support for capturing frames from real video devices such
+	// as a webcam on /dev/video0)
+	IMAGE_PICTURE(config, m_input1);
+	IMAGE_PICTURE(config, m_input2);
+	IMAGE_PICTURE(config, m_input3);
+	IMAGE_PICTURE(config, m_input4);
+
+	SCREEN(config, m_screen, SCREEN_TYPE_RASTER);
```

Comment:
If the service manual provides appropriate timing values, please use them via `set_raw`, as the combination of `set_refresh_hz`, `set_size`, and `set_visarea` are deprecated. If the raw timing values are not available, it's fine as it is.

Reply:
I haven't found that info yet. Maybe the service manual provides it, but I haven't seen it. It is a very long manual! :-)

If I ever find the info, I'll update the driver for sure